addons/gearinfo/functions/fnc_optionValues.sqf

```diff
@@ -0,0 +1,33 @@
+#include "script_component.hpp"
+
+params ["_classRoot", "_model", "_optionName"];
+
+private _modelDefinition = configFile >> "XtdGearModels" >> _classRoot >> _model;
+
+private _optionValues = GVAR(optionValueCache) getOrDefault [[_classRoot, _model, _optionName], []];
+
+if (count _optionValues != 0) exitWith {
+    _optionValues
+};
+
+private _optionDef1 = _modelDefinition >> _optionName;
+private _optionDef2 = configFile >> "XtdGearModels" >> "Conventional" >> _optionName;
+
+private _definedOptionValues = [_optionDef1, _optionDef2, "values", []] call READ_ARRAY;
+
+_optionValues = if (count _definedOptionValues > 0) then { _definedOptionValues } else {
+    private _classPaths = [_optionDef1, _optionDef2];
+    {
+        private _baseClassPath = inheritsFrom _x;
+        if (not isNull _baseClassPath) then {
+            _classPaths append [_baseClassPath];
+        };
+    } forEach _classPaths;
+    private _subClassNames = _classPaths apply { _x call Bis_fnc_getCfgSubClasses };
+    private _allValues = flatten _subClassNames;
+    _allValues arrayIntersect _allValues;
+};
+
+GVAR(optionValueCache) set [[_classRoot, _model, _optionName], _optionValues];
```

Comment:
since everything will be cached at getModelOptions level, this cache level is not required